src/dqn_agent.py

```diff
@@ -17,14 +19,48 @@
 EPSILON_END = 0.01
 EPSILON_STEPS = 10000
 
+VIDEO_INTERVAL = 10000
 
-# first baseline
-def q_model(x, n):
-    h1 = tflearn.fully_connected(x, 20, activation='relu')
-    out = tflearn.fully_connected(h1, n)
+
+def single_agent_model(x, n_actions):
+    with tf.name_scope("single_agent"):
+        h1 = tflearn.fully_connected(x, 20, activation='relu')
+        out = tflearn.fully_connected(h1, n_actions)
+        return out
+
+
+def sequential_model(x, n_agents, n_actions):
+    qs = [single_agent_model(x, n_actions)]
+    smaxes = [tflearn.activations.softmax(qs[-1])]
+
+    for i in range(1, n_agents):
+        smaxes.append(x)
+        x_i = tf.concat(smaxes, axis=1)
+        smaxes.remove(x)
```

Comment:
maybe instead of appending and removing, do can do x_i = tf.concat(smaxes + [x], axis =1)